spec/rubocop/cop/style/redundant_conditional_spec.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+describe RuboCop::Cop::Style::RedundantConditional do
+  let(:config) { RuboCop::Config.new }
+  subject(:cop) { described_class.new(config) }
+
+  before { inspect_source(cop, source) }
+
+  shared_examples 'code with offense' do |code, expected, message_expression|
+    context "when checking #{code}" do
+      let(:source) { code }
+
+      it 'registers an offense' do
+        expected_message =
+          'This conditional expression '\
+          "can just be replaced by #{message_expression || expected}"
+        expect(cop.offenses.size).to eq(1)
+        expect(cop.messages).to eq([expected_message])
+      end
+
+      it 'auto-corrects' do
+        expect(autocorrect_source(cop, code)).to eq(expected)
+      end
+
+      it 'claims to auto-correct' do
+        autocorrect_source(cop, code)
+        expect(cop.offenses.last.status).to eq(:corrected)
+      end
+    end
+  end
+
+  shared_examples 'code without offense' do |code|
+    let(:source) { code }
+
+    it 'does not register an offense' do
+      expect(cop.offenses).to be_empty
+    end
+  end
+
+  it_behaves_like 'code with offense',
+                  'x == y ? true : false',
+                  'x == y'
+
+  it_behaves_like 'code with offense',
+                  'x == y ? false : true',
+                  '!(x == y)'
+
+  it_behaves_like 'code without offense',
+                  'x == y ? 1 : 10'
+
+  it_behaves_like 'code with offense',
+                  <<-END.strip_indent,
+                    if x == y
+                      true
+                    else
+                      false
+                    end
+                  END
+                  "x == y\n",
+                  'x == y'
```

Comment:
@bbatsov not too happy with these specs now, they feel kind of messy with the trailing newline and then a repeat for the message. Any thoughts?

Could assume the message will be the expression stripped of any trailing newlines but that's not guaranteed to be true.